solenspipe/solenspipe.py

```diff
@@ -304,13 +326,65 @@ def prepare_map(self,channel,seed,lmin,lmax):
         # Cache the alms
         self.cache = {}
         self.cache[seed] = (almt,alme,almb,oalms[0],oalms[1],oalms[2])
+        icov,s_set,i=seed
+        
+    def get_sim_power(self,channel,seed,lmin,lmax):
```

Comment:
@qujia7 can you say a bit about what these new functions are for?

Reply:
This function generates the beam deconvolved total Cls from simulations used by the diagonal_rdn0 functions.

Reply:
This function calculates the beam deconvolved total Cls from simulations to be used in the diagonal RDN0 functions.